### PR TITLE
Upgrade to v2.25.0 of the AzureRM Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #678
Fixes: #680

### CODE REVIEW

File: provider/resource.go [hunk 1]
```diff
@@ -593,18 +594,42 @@ func Provider() tfbridge.ProviderInfo {
...
+			"azurerm_snapshot":             {Tok: azureResource(azureCompute, "Snapshot")},
+			"azurerm_image":                {Tok: azureResource(azureCompute, "Image")},
+			"azurerm_shared_image":         {Tok: azureResource(azureCompute, "SharedImage")},
+			"azurerm_shared_image_gallery": {Tok: azureResource(azureCompute, "SharedImageGallery")},
+			"azurerm_shared_image_version": {Tok: azureResource(azureCompute, "SharedImageVersion")},
+			"azurerm_bastion_host":         {Tok: azureResource(azureCompute, "BastionHost")},
+			"azurerm_dedicated_host_group": {Tok: azureResource(azureCompute, "DedicatedHostGroup")},
+			"azurerm_disk_encryption_set":  {Tok: azureResource(azureCompute, "DiskEncryptionSet")},
+			"azurerm_dedicated_host":       {Tok: azureResource(azureCompute, "DedicatedHost")},
+			"azurerm_linux_virtual_machine": {
```
Comment (mikhailshilkov) [inline provider/resource.go:607]: Add a link to https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftcompute

**Review by mikhailshilkov — APPROVED** [review]
LGTM

File: provider/resource.go (hunk 1 above)
Comment (stack72) [reply provider/resource.go:607]: Ok, will add this locally when merging

**Comment by stack72** [comment]
Merged locally to add a comment for the size of the compute resource name

**Pull request closed without merging**